### PR TITLE
Added internationalization for file filter dropdown

### PR DESCRIPTION
**Pull request by sahariardev**
Summery

Made Files filter dropdown menu internationalized.

Release Note
```release-note
NONE
```

Fixes following issue:
Fixes https://github.com/mattermost/mattermost/issues/28648

### CODE REVIEW

File: webapp/channels/src/i18n/en.json [hunk 1]
```diff
@@ -5893,5 +5893,14 @@
   "workspace_limits.teams_limit_reached.view_upgrade_options": "View upgrade options",
   "workspace_limits.upgrade": "Upgrade to avoid {planName} data limits",
   "workspace_limits.upgrade_reasons.free": "{planName} is restricted to {messagesLimit} message history and {storageLimit} file storage. You can delete items to free up space or upgrade to a paid plan.",
-  "yourcomputer": "Your computer"
+  "yourcomputer": "Your computer",
+  "channel_info_rhs.menu.files.filter.file_menu" : "file menu",
```
Comment (Willyfrog) [inline webapp/channels/src/i18n/en.json:5897]: I think this should be moved up as it is expected to be in alphanumeric order. It's likely the reason why it is failing the test

**Review by Willyfrog — CHANGES_REQUESTED** [review]
Thanks for the submission, but unfortunately the tests aren't passing. Can you take a look at them?

File: webapp/channels/src/i18n/en.json (hunk 1 above)
Comment (sahariardev) [reply webapp/channels/src/i18n/en.json:5897]: will fix them

**Comment by sahariardev** [comment]
> Thanks for the submission, but unfortunately the tests aren't passing. Can you take a look at them?

Hi @Willyfrog 
I have made those changes. Please let me know anything else I need to do.

**Comment by Willyfrog** [comment]
Awesome, I've started the tests again :)

**Comment by sahariardev** [comment]
hi @Willyfrog,
I am not sure why this failing again, could you please guide me with this?

**Comment by Willyfrog** [comment]
> hi @Willyfrog, I am not sure why this failing again, could you please guide me with this?

I think it is because you have a space before the colon. I mean `" : "` should look like `": "`

**Comment by sahariardev** [comment]
> > hi @Willyfrog, I am not sure why this failing again, could you please guide me with this?
> 
> I think it is because you have a space before the colon. I mean `" : "` should look like `": "`

Thanks @Willyfrog. I have update the commit

**Review by Willyfrog — APPROVED** [review]
LGTM, thanks!

**Comment by Rutam21** [comment]
@lindy65 Please assign this PR to me for QA Testing.

**Comment by lindy65** [comment]
Assigned to you for testing @Rutam21 :)

**Comment by Rutam21** [comment]
### This PR has been tested and the QA Test has failed as the intended changes are not visible in the Test Server.

- When the server display language is changed to something other than English, the File Search Filter dropdown options still show in English only.

Please take a look at the snapshots below as a confirmation.

#### Server Language: Deutsch

PRTestServerDeutsch

#### Server Language: Turkce

PRTestServerTurkce

#### Server Language: Espanol Beta

PRTestServerEspanolBeta


cc: @lindy65

**Comment by lindy65** [comment]
@Willyfrog - please take a look at @Rutam21's testing [here](https://github.com/mattermost/mattermost/pull/28809#issuecomment-2429802029)
Are these results correct? I'm not familiar with translation expectations

**Comment by cwarnermm** [comment]
Given that these are new product strings, they don't yet have translations into any officially supported language. As a result, Mattermost will continue to display these new strings, regardless of the client language, in US English.

I suspect we'd need to test that Mattermost will serve up translations, where available. To do this, I think we could manually modify one of the language JSON files, change the client language to match, and confirm that Mattermost serves up the right strings.

For example, if we had a test environment, and modified the fr.json language file to include these new strings, modified the string translation from US English to a test string, and switched the MM client to French, would Mattermost serve up that test string successfully?

What I don't know is whether a test env spun up gives us the access we need to the language JSON files to perform such a test. @Willyfrog - do you know if this testing approach would be valid?

**Comment by Willyfrog** [comment]
@cwarnermm That approach would definitely work in a self-hosted test. You don't even need to know the language. Adding a random string and seeing it appear would be a green check mark.

I don't think we have the option to inject it into the Spinwick server. That means that the person testing something like this would need a basic knowledge of how to set up Mattermost. However, I think it would work for specific cases like this, and I don't expect a lot of them, so there is little incentive to set up some tooling to help.

**Comment by lindy65** [comment]
Thanks @Willyfrog - I'm inclined to merging this PR and testing post merge once translation strings are done.

cc @Rutam21 for your info

**Comment by Willyfrog** [comment]
That's fine with me

**Review by lindy65 — APPROVED** [review]
Thanks for the contribution @sahariardev 🙌 

Approving on behalf of QA to test post merge / once translation strings are completed

As you did test this @Rutam21 and saw the translation strings were not completed, it is counted towards Hacktoberfest

**Review by rahimrahman — APPROVED** [review]
Approving.  Thanks for your contribution.

A few nits (based on my own personal coding practice of DRY) so it is not required for you to make changes to your work.  This is in the existing code, but in the future, I encourage simplifying by eliminating code repetition:

1) Consider de-structuring objects:

```
const {intl: {formatMessage}} = useIntl();
const {onClick, selectedFilter} = props;
```

2. And creating a new internal component within this file:

```
type MenuItemActionProps {
  filter: string;
  intlMessage: Intl.SomeIntlMessageType;
  onClick: () => void;
  isSelected: boolean;
};

const MenuItemAction = ({filter, intlMessage, onClick, isSelected}: MenuItemActionProps) => {
  const handleOnClick = () => {
    onClick(filter);
  };
  return (
    <Menu.ItemAction
                        ariaLabel={intlMessage}
                        text={intlMessage}
                        onClick={handleOnClick}
                        icon={isSelected ? <i className='icon icon-check'/> : null}
                    />
  )
}
```

  then all can be simplified:

```
  <MenuItemAction filter={'code'}
      intlMessage={formatMessage({id: 'channel_info_rhs.menu.files.filter.code', defaultMessage: 'Code'})
      onClick={onClick}
      isSelected={selectedFilter === 'code'} />
```

3) Could even go further by making an array of objects and looping the `<MenuItemAction />`, for even less code repetition.

Again, this is personal preference.

**Comment by hanzei** [comment]
@rahimrahman Are you comments blocking or can we merge this PR?

**Pull request merged**